src/lrucache.h

```diff
@@ -43,10 +43,18 @@
 #include <cstddef>
 #include <stdexcept>
 #include <cassert>
+#include <vector>
 
 namespace zim {
 
-template<typename key_t, typename value_t>
+struct UnitCostEstimation {
+  template<typename value_t>
+  static size_t cost(const value_t& value) {
+    return 1;
+  }
+};
+
+template<typename key_t, typename value_t, typename CostEstimation>
 class lru_cache {
```

Comment:
This is a serious enhancement that should be difficult to do right. LRU is about heuristics of maintaining items in the cache based on the order they have been used. Introducing cost adds an important dimension to consider - when facing the choice of dropping from the cache an expensive old item or a cheap not-so-old one, which one is more justified? Or the other way around - it may be preferable to get rid of a heavy item that is *not* the first candidate for removal according to the LRU heuristics since that will make more room in the cache for multiple lightweight items.